packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1692,6 +1719,23 @@ export class Bucket extends BucketBase {
       };
     });
   }
+
+  private getOrCreateAutoDeleteObjectsResource() {
+    return CustomResourceProvider.getOrCreate(this, AUTO_DELETE_OBJECTS_RESOURCE_TYPE, {
+      codeDirectory: path.join(__dirname, 'auto-delete-objects-handler'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [
+        {
+          Effect: 'Allow',
+          Resource: '*',
+          Action: [
+            // TODO: add key perms?
+            ...perms.BUCKET_READ_ACTIONS.concat(perms.BUCKET_WRITE_ACTIONS),
```

Comment:
```suggestion
            ...perms.BUCKET_READ_ACTIONS,
            ...perms.BUCKET_DELETE_ACTIONS,
```

Reply:
I suggested `BUCKET_DELETE_ACTIONS`, wasn't it sufficient?

Reply:
You're right, I totally missed that - thanks!